src/compiler/checker.ts

```diff
@@ -6609,7 +6611,7 @@ module ts {
             }
 
             // The identityMapper object is used to indicate that function expressions are wildcards
-            if (contextualMapper === identityMapper) {
+            if (contextualMapper === identityMapper && isContextSensitive(node)) {

```

Comment:
I assume you're keeping the `contextualMapper === identityMapper` as a quick short-circuiting operation.